tests/typecheck/managers/querysets/test_values.yml

```diff
@@ -124,3 +124,21 @@
                     pass
                 class Book(models.Model):
                     authors = models.ManyToManyField(Author, related_name='books')
+
+-   case: queryset_values_blank_charfield
+    main: |
+        from myapp.models import Blog
+        values = Blog.objects.values('text').get()
```

Comment:
Forgive me my lack of knowledge here, but what will happen in runtime?

`None` -> `''`?

Reply:
I'm not sure if I'm fully understanding what you're asking, but setting `blank=True` on a field does not affect the nullability of that field in the database. All it does is enable a Django-level check to either allow or disallow empty strings to be saved. That means that in the database the field will not accept null values unless `null=True`. An empty string will be stored as just that, and empty string, and when querying the database it will always return a string value, either empty or not.

This plugin will however return blank CharFields as `Optional[str]`, even if that's never gonna happen at runtime because the database (at least a strictly type checking one like Postgres, not sure about SQLite) will never allow you to store null values in the column

Reply:
Thank you, I get it now! 👍